docs/reference/src/main/asciidoc/configure.asciidoc

```diff
@@ -273,6 +273,37 @@ The functionality is implemented as a built-in portable extension processing all
 |`org.jboss.weld.bootstrap.vetoTypesWithoutBeanDefiningAnnotation` || A regular expression. If a non-empty string, then all annotated types whose `AnnotatedType#getJavaClass().getName()` matches this pattern are vetoed if not annotated with a bean defining annotation.
 |=======================================================================
 
+[[remove-unused-beans]]
+==== Memory consumption optimization - removing unused beans
+
+CDI applications consist of user-defined beans implementing the business logic, but also beans coming from libraries (e.g. DeltaSpike) and integrations (e.g. various Java EE technologies - JSF, Bean Validation, Batch).
+Most applications actually use only a small part of the beans provided by libraries and integrations.
+Still, Weld has to retain the metadata for all beans in memory!
+What a waste.
+If <<allow-optimized-cleanup>> is allowed Weld can now remove the _unused_ beans after bootstrap.
+
+An _unused_ bean:
+
+* is not excluded by the configuration properties described below,
+* is not a built-in bean, session bean, extension, interceptor or decorator,
+* does not have a name
+* does not declare an observer
+* is not eligible for injection to any injection point,
+* does not declare a producer which is eligible for injection to any injection point,
+* is not eligible for injection into any `Instance` injection point.
```

Comment:
I suppose you only mean an IP of type `Instance<X>` where X is the potentially unused bean. However, this sentence sounds a bit like it's a bean which you cannot even `Instance.select()` (which is untrue as we have no power to detect that). Adding an example in brackets may make it clearer?

Reply:
Well, `Instance.select()` is not an `Instance` injection point. I mean any `@Inject Instance<X>` which can potentially resolve to the bean. And if you do `Instance.select()` afterwards it's ok because you can only select subtype.

Reply:
Alright, if you think it's obvious enough, then leave it as it is.

Reply:
We can always explain to a few users on gitter or so... ;-)